### PR TITLE
tests: arch: arm: improve filtering to fix several issues with platforms that shall not run these tests

### LINKED ISSUE #28294: arch.interrupt.gen_isr_table.arm_mainline: lpcxpresso55s16_ns failed
**Describe the bug**
interrupt 57 didn't run once, ran 0 times

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=lpcxpresso55s16_ns ..
3. make

**Expected behavior**
test pass

**Impact**
core interrupt settings

**Logs and console output**

```
 *** Booting Zephyr OS version 2.4.0-rc1  ***

START - Test gen_isr_tables

IRQ configuration (total lines 60):

isr1 isr=0x000004b9 irq=59

isr2 isr=0x000004e1 irq=58

Checking _irq_vector_table entry 59 for irq 59

triggering irq 59

isr1 ran

Checking _irq_vector_table entry 58 for irq 58

triggering irq 58

isr2 ran

_sw_isr_table at location 0x20000018

isr3 isr=0x00000429 irq=57 param=0xb01dface

Checking _sw_isr_table entry 57 for irq 57

triggering irq 57

interrupt 57 didn't run once, ran 0 times

 FAIL - main

===================================================================

===================================================================

PROJECT EXECUTION FAILED

```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v2.4.0-rc1

### PR DESCRIPTION
**Pull request by ioannisg**
- Do some cleaning in the .yml files for tests/arch/arm
- Do not attempt to run or build the tests for strictly non-secure platform configurations.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/28294

### CODE REVIEW

File: tests/arch/arm/arm_interrupt/testcase.yaml [hunk 1]
```diff
@@ -1,19 +1,14 @@
+common:
+  filter: CONFIG_ARMV6_M_ARMV8_M_BASELINE or CONFIG_ARMV7_M_ARMV8_M_MAINLINE
+  tags: arm interrupt ignore_faults
+  arch_allow: arm
```
Comment (MaureenHelm) [inline tests/arch/arm/arm_interrupt/testcase.yaml:4]: isn't this redundant to the filter?

Comment (ioannisg) [reply tests/arch/arm/arm_interrupt/testcase.yaml:4]: Not sure (@nashif to confirm)
I think the arch allow (earlier: whitelist) is required for proper filtering application

Comment (ioannisg) [reply tests/arch/arm/arm_interrupt/testcase.yaml:4]: All tests/arch/arm have the arch_allow: arm listed.

This PR does not change that. It is only grouping together common properties, in the test variants, and adds the !TRUSTED_EXECUTION_NONSECURE filter when necessary.

**Review by hakehuang — APPROVED** [review]

File: tests/arch/arm/arm_interrupt/testcase.yaml (hunk 1 above)
Comment (MaureenHelm) [reply tests/arch/arm/arm_interrupt/testcase.yaml:4]: I understand you're just refactoring. Seems worth cleaning up the redundancy while we're here though. @nashif, do we need the arch_allow when we're filtering on arm-specific kconfigs?

Comment (ioannisg) [reply tests/arch/arm/arm_interrupt/testcase.yaml:4]: @MaureenHelm, cannot make this work without the arch_allow directive; must be something with the sanity check filtering algorithm that prevents from successfully filtering-out platforms.

Comment (MaureenHelm) [reply tests/arch/arm/arm_interrupt/testcase.yaml:4]: ok, we can leave it as-is then

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**